src/connectors/hits-per-page/connectHitsPerPage.ts

```diff
@@ -235,11 +220,20 @@ You may want to add another entry to the \`items\` option with this value.`
         };
       },
 
-      getWidgetRenderState({ state, results }) {
+      getWidgetRenderState({ state, results, createURL }) {
+        connectorState.createURLFactory = helperState => value => {
+          return createURL(
+            helperState.setQueryParameter(
+              'hitsPerPage',
+              !value && value !== 0 ? undefined : value
+            )
+          );
+        };
```

Comment:
I think this function should probably stay in init for consistency with connectorState.setHitsPerPage, but I'm not sure how other connectors dealt with this yet

Reply:
I agree. It can stay in `init`.

Reply:
basically the problem is that in the test (similar to breadcrumb) it seems to call getWidgetRenderState before init has happened, which means it will throw, since connectorState is still empty object at that point